.github/workflows/rust.yml

```diff
@@ -179,6 +179,59 @@ jobs:
                     --arg ddtags "$tags" \
                     '{ message: $message, ddsource: "hash-gh-actions", hostname: "github", service: "actions", ddtags: $ddtags }')
 
+  coverage:
+    name: coverage
+    needs: setup
+    if: needs.setup.outputs.coverage != '{}'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.setup.outputs.coverage) }}
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
```

Comment:
Can we use the same action everywhere? `action-rs/toolchain@v1` is [unmaintained](https://github.com/actions-rs/toolchain/issues/216). We could go for `./.github/actions/install-rust-toolchain` or some third-party alternative.

Reply:
I'd prefer doing this to be done in a follow up as this PR is unrelated